internal/provider/resource_vault_secrets_app.go

```diff
@@ -0,0 +1,168 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	clients "github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func NewVaultSecretsAppResource() resource.Resource {
+	return &resourceVaultsecretsApp{}
+}
+
+type resourceVaultsecretsApp struct {
+	client *clients.Client
+}
+
+func (r *resourceVaultsecretsApp) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_secrets_app"
+}
+
+func (r *resourceVaultsecretsApp) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The Vault Secrets app resource manages an application.",
```

Comment:
Perhaps would be better to say something like "... manages secrets for an application."

Reply:
This particular resource only allows you to manage an application. And this gets generated in specific docs for this specific resource.